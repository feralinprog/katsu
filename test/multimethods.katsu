use: "example.imports.indirect"

generic: (n: Fixnum) do-a-thing

use: "example.imports.multimethods"

defer: (n: Fixnum) do-another-thing

5 do-a-thing
10 do-another-thing

# Didn't import core.builtin.misc!
generic: (seq append: value)
let: v = {}
v append: 15

use: "core.builtin.misc"
pretty-print: v
