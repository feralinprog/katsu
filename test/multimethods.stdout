doing a thing!
fixnum 5
doing another thing!
fixnum 10
*vector: length=1 [
  v_array = *array: length=2
    0 = fixnum 15
    1 = null
]
